google-cloud-firestore/samples/add_data.rb

```diff
@@ -72,6 +72,54 @@ def set_document_data_types project_id:, collection_path: "data"
   puts "Set multiple data-type data for the one document in the data collection."
 end
 
+# [START firestore_data_custom_type_definition]
+class City
+  attr_accessor :name, :state, :country, :capital, :population, :regions
+
+  def initialize name, state, country, capital: false, population: 0, regions: []
+    @name = name
+    @state = state
+    @country = country
+    @capital = capital
+    @population = population
+    @regions = regions
+  end
+
+  def self.from_h source
+    city = new source[:name], source[:state], source[:country]
+    city.capital = source[:capital] if source[:capital]
+    city.population = source[:population] if source[:population]
+    city.regions = source[:regions] if source[:regions]
```

Comment:
These should probably be, e.g. `unless source[:capital].nil?` or `if source.key? :capital`

Reply:
Fixed, thanks.